src/api/structure/test.js

```diff
@@ -13,17 +13,19 @@ export default class Test extends TestingUnit {
         this.fn           = null;
         this.beforeFn     = null;
         this.afterFn      = null;
-        this.requestHooks = this.fixture.requestHooks.length ? Array.from(this.fixture.requestHooks) : [];
+        this.requestHooks = [];
 
         return this.apiOrigin;
     }
 
     _add (name, fn) {
         assertType(is.string, 'apiOrigin', 'The test name', name);
         assertType(is.function, 'apiOrigin', 'The test body', fn);
+        assertType(is.nonNullObject, 'apiOrigin', `The fixture of '${name}' test`, this.fixture);
 
-        this.name = name;
-        this.fn   = wrapTestFunction(fn);
+        this.name         = name;
+        this.fn           = wrapTestFunction(fn);
+        this.requestHooks = union(this.requestHooks, Array.from(this.fixture.requestHooks));
```

Comment:
As I remember, we can use `Array.from(this.fixture.requestHooks)` for cloning request hooks.
```js
this.requestHooks = Array.from(this.fixture.requestHooks);
```

Reply:
I'm using it. I just need `union` to merge both `test.requestHooks` and `fixture.request.hooks`

Reply:
Ok